cmd/kube-proxy/app/server.go

```diff
@@ -110,87 +109,110 @@ func checkKnownProxyMode(proxyMode string) bool {
 	return false
 }
 
-// Run runs the specified ProxyServer.  This should never exit (unless CleanupAndExit is set).
-func (s *ProxyServer) Run(_ []string) error {
+func NewProxyConfig() *ProxyServerConfig {
+	return &ProxyServerConfig{
+		BindAddress:        net.ParseIP("0.0.0.0"),
+		HealthzPort:        10249,
+		HealthzBindAddress: net.ParseIP("127.0.0.1"),
+		OOMScoreAdj:        qos.KubeProxyOomScoreAdj,
+		ResourceContainer:  "/kube-proxy",
+		SyncPeriod:         5 * time.Second,
+	}
+}
+
+func NewProxyServer(
+	config *ProxyServerConfig,
+	client *kubeclient.Client,
+	endpointsConfig *proxyconfig.EndpointsConfig,
+	endpointsHandler proxyconfig.EndpointsConfigHandler,
+	iptInterface utiliptables.Interface,
+	oomAdjuster *oom.OomAdjuster,
+	proxier proxy.ProxyProvider,
+	recorder record.EventRecorder,
+	serviceConfig *proxyconfig.ServiceConfig) (*ProxyServer, error) {
```

Comment:
it's conventional to add a trailing comma and line break before the closing paren